aptos-move/framework/aptos-framework/sources/account_abstraction/permissioned_delegation.move

```diff
@@ -0,0 +1,190 @@
+module aptos_framework::permissioned_delegation {
+    use std::error;
+    use std::option::Option;
+    use std::signer;
+    use aptos_std::ed25519;
+    use aptos_std::ed25519::{new_signature_from_bytes, new_unvalidated_public_key_from_bytes, UnvalidatedPublicKey};
+    use aptos_std::table::{Self, Table};
+    use aptos_framework::auth_data;
+    use aptos_framework::auth_data::AbstractionAuthData;
+    use aptos_framework::bcs_stream;
+    use aptos_framework::bcs_stream::deserialize_u8;
+    use aptos_framework::permissioned_signer::{Self, is_permissioned_signer, StorablePermissionedHandle};
+    use aptos_framework::token_bucket;
+    #[test_only]
+    use std::bcs;
+    #[test_only]
+    use std::option;
+
+    const ENOT_MASTER_SIGNER: u64 = 1;
+    const EINVALID_PUBLIC_KEY: u64 = 2;
+    const EPUBLIC_KEY_NOT_FOUND: u64 = 3;
+    const EINVALID_SIGNATURE: u64 = 4;
+    const EHANDLE_EXISTENCE: u64 = 5;
+    const ERATE_LIMITED: u64 = 6;
+
+    enum HandleBundle has store {
+        V1 { handle: StorablePermissionedHandle, bucket: Option<token_bucket::Bucket> }
+    }
+
+    struct Delegation has key {
+        handle_bundles: Table<ed25519::UnvalidatedPublicKey, HandleBundle>
```

Comment:
hm should we be using public_key here, or authentication_key (i.e. what we use as account address) - the later seems more intuitive to me.

Reply:
here the authentication is made through a ed25519 key pair, which has nothing to do with aptos account. we can do the authentication_key version, which require the whole signing scheme, authentication_key, public key, signing method (multisig, single_key and which cryptographic algorithm... which is way more complicated)

Reply:
maybe to make this potentially expandable later, change the key here to :

```
enum HandleKey {
   Ed25519PublicKey(ed25519::UnvalidatedPublicKey)
 }
```

so that you have 
handle_bundles: Table<HandleKey, HandleBundle>

you can then also change add_permissioned_handle to receive HandleKey instead of bytes?